library/user.inc

```diff
@@ -9,139 +9,123 @@
+ * 
+ * @param string $label - Setting key
+ * @param string $value - Setting value
+ * @param int $user - user id number from users table
+ * @param boolean $createDefault - If no current global default value, create one. 
+ */
 function setUserSetting($label, $value, $user=NULL, $createDefault=TRUE) {
-  // if no user id is sent, then use the currently logged in user
-  if (!isset($user)) {
-    $user = $_SESSION['authUserID'];
-  }
+  
+  $user = effectiveUser($user);
 
-  $newCustom=false;
-  $newDefault=false;
+  $cur_value = getUserSetting($label, $user);
 
   // Check for a custom settings
-  $row = sqlQuery("SELECT setting_value FROM user_settings " .
-    "WHERE setting_user=? AND setting_label=?", array($user, $label) );
-  if (empty($row)) {
-    $newCustom=true;
-  }
-
-  // Set the custom setting
-  if ($newCustom) {
-    sqlStatement("INSERT INTO user_settings " .
-      "(setting_user, setting_label, setting_value) " .
+  if (is_null($cur_value)) {
+    sqlStatement("INSERT INTO user_settings(setting_user, setting_label, setting_value) " .
       "VALUES (?,?,?)", array($user, $label, $value) );
-  }
-  else {
+  } elseif ($cur_value !== $value) {
     sqlStatement("UPDATE user_settings SET setting_value=? " .
       "WHERE setting_user=? AND setting_label=?", array($value, $user, $label) );
   }
-
-  // Check for a default settings
-  $row = sqlQuery("SELECT setting_value FROM user_settings " .
-    "WHERE setting_user='0' AND setting_label=?", array($label) );
-  if (empty($row)) {
-    $newDefault=true;
-  }
-
-  // Place a default setting if it does not already exist (and createDefault flag is set)
-  if ($newDefault && $createDefault) {
-    sqlStatement("INSERT INTO user_settings " .
-      "(setting_user, setting_label, setting_value) " .
-      "VALUES ('0',?,?)", array($label, $value) );
+  
+  // Call self to create default value
+  if ($createDefault) {
+    setUserSetting($label, $value, 0, false);
   }
 }
 
 //This will remove the selected user setting from the 'user_settings' table.
 // $label is used to determine which setting to remove
-// $user is the user id number in the 'users' table
+// $user is the user id number from users table
 function removeUserSetting($label,$user=NULL) {
-  // if no user id is sent, then use the currently logged in user
-  if (!isset($user)) {
-    $user = $_SESSION['authUserID'];
-  }
+
+  $user = effectiveUser($user);
   
-  // Collect entry if it exist.
-  //  If it exists, then delete the entry.
-  $result = sqlQuery("SELECT setting_value FROM user_settings " .
-    "WHERE setting_user=? AND setting_label=?", array($user, $label) );
-  if (!empty($result)) {
-    sqlQuery("DELETE FROM user_settings " .
+  // mdsupport - DELETE has implicit select, no need to check and delete
+  sqlQuery("DELETE FROM user_settings " .
       "WHERE setting_user=? AND setting_label=?", array($user, $label) );
-  }
 }
 
-
+// mdsupport - Seems superfluous - sqlQuery function will do the same thing.
```

Comment:
Just a philosophical peeve about single line functions used in a single place without any transformation - specially since neither this or the caller function notify its caller if users.id is not found.

Will let it lie.